dev/TeachingTip/TeachingTip.cpp

```diff
@@ -894,29 +927,37 @@ void TeachingTip::OnTailOcclusionGridLoaded(const winrt::IInspectable&, const wi
 }
 
 
-void TeachingTip::RaiseClosingEvent()
+void TeachingTip::RaiseClosingEvent(bool attachDeferralCompletedHandler)
 {
     auto args = winrt::make_self<TeachingTipClosingEventArgs>();
     args->Reason(m_lastCloseReason);
 
-    com_ptr<TeachingTip> strongThis = get_strong();
-    winrt::DeferralCompletedHandler instance{ [strongThis, args]()
-        {
-            strongThis->CheckThread();
-            if (!args->Cancel())
-            {
-                strongThis->ClosePopupWithAnimationIfAvailable();
-            }
-            else
+    if (attachDeferralCompletedHandler)
+    {
+        com_ptr<TeachingTip> strongThis = get_strong();
+        winrt::DeferralCompletedHandler instance{ [strongThis, args]()
             {
-                // The developer has changed the Cancel property to true, indicating that they wish to Cancel the
-                // closing of this tip, so we need to revert the IsOpen property to true.
-                strongThis->IsOpen(true);
+                strongThis->CheckThread();
+                if (!args->Cancel())
+                {
+                    strongThis->ClosePopupWithAnimationIfAvailable();
+                }
+                else
+                {
+                    // The developer has changed the Cancel property to true, indicating that they wish to Cancel the
+                    // closing of this tip, so we need to revert the IsOpen property to true.
+                    strongThis->IsOpen(true);
+                }
             }
-        }
-    };
+        };
 
-    args->SetDeferral(instance);
+        args->SetDeferral(instance);
+    }
+    else
+    {
+        winrt::DeferralCompletedHandler dummyInstance{ []() {} };
+        args->SetDeferral(dummyInstance);
+    }
```

Comment:
Why do we need to do this? Needs comment. I also don't understand what this parameter is for.

Reply:
I'll change this to be a little less obtuse, but in the case where the tip is not opening because it does not fit, we still need to raise the closing and closed events. However, we can't delay the closing of the popup until after the closing event's deferral has completed since it is effectively already closed.  This is an attempt to raise an event args which still has a deferral object but which doesn't do anything, @Jevan Saks do you think it is better for the deferral in the args to be null or dummy?

---
In reply to: [268775343](https://github.com/Microsoft/microsoft-ui-xaml/pull/480#discussion_r268775343) [](ancestors = 268775343)